### DISCUSSION TITLE
Limit transit walking distance for isochrone

### DISCUSSION THREAD

**Discussion by khamaileon** [valhalla/valhalla#4163] [Q&A] [answered]
Here are 2 isochrones. The first is on foot, with 6 intervals of 10 minutes each. The second use transit mode, but without a loaded GTFS. In the second isochrone, you can see that it stalls at around 35mn. I imagine this is a limit on walking to a station. However, I can't find this parameter in the config. I've tried max_transit_walking_distance but it only applies to the route endpoint.

![image](https://github.com/valhalla/valhalla/assets/1322166/1e9fdd90-6792-4fc5-bd66-a1e9541cdafb)
![image](https://github.com/valhalla/valhalla/assets/1322166/c226acec-38b9-4e1b-a992-e1631bbad1f5)

**Comment by khamaileon** [comment]
I found something interesting in the valhalla-app project but it doesn't seem to be used 🤔

![image](https://github.com/valhalla/valhalla/assets/1322166/b7e39143-c839-4d6c-8d5a-048161be9a98)

**Reply by nilsnolde** [reply, marked as answer]
Jep, you're on the right track.. In the docs, a bit confusingly, but technically correct, quite a few multimodal options are listed under pedestrian. However, `max_transit_walking_distance` is a static config that'll generally limit the `transit_start_end_max_distance` request option. So that's the one you'll want to tweak per request. It'll default to ~2 km, and IIRC you can only change the default in code.

**Reply by nilsnolde** [reply]
reg the valhalla app: all costing options were initially listed, but we don't have a concept for inputting time constraints (yet), so no point for multimodal options. Eventually (within 2023 :crossed_fingers:), I'd like to charge the public instance with some GTFS, but unpaid time is not so easy these days.. Also the prospect of very few feeds taking days to compute is just not very sexy to me at the moment. On the other hand, taking the (pretty long) time to optimize the transit build pipeline is not an option either for the foreseeable future. Meeh, never mind me brooding here, I'll reason about it again on a less gloomy day :sweat_smile:

**Reply by moss-xyz** [reply]
@nilsnolde adding on to this as I have a question that is related (I think).

I'm currently trying to find a route between two points. It works in `pedestrian` mode, but doesn't in `multimodal` (returns a "no path found" error). Are you saying that I should be able to append something like `"transit_start_end_max_distance":10.0` to the `"costing_options":{"transit":{}}` dict to increase the search radius to the nearest bus stop, if I have a hunch it is just too far away to find (you say the default is 2km whereas I think the nearest bus stop is 2 mi)? I just tried monkeying around with that and it didn't seem to work.

**Reply by nilsnolde** [reply]
You should have the transit data locally right? 

The costing option is in the pedestrian namespace, not transit. It’s not very intuitive, albeit correct. Best to look at the docs.

**Reply by moss-xyz** [reply]
Ah whoops! Totally missed that in the docs, my bad, sorry! It doesn't seem to have fixed my issue anyways, weirdly.

Maybe I'm having a weird issue with either my start or end location being located in a small, disconnected island? Any issue on determining that and fixing it?

**Reply by moss-xyz** [reply]
@nilsnolde interestingly, I seemed to have fixed my own issue by lowering the "minimum_reachability" threshold from 50 to 5.

Any context on why this works? I'm confused because keeping the threshold at 50 works for pedestrian and auto modes, but specifically _not_ for the bicycle mode...

**Reply by nilsnolde** [reply]
Bicycle? Multimodal? I guess typo and the latter?

Is that common in your network? Can you show us the logs of the transit build pipeline?

not sure what it’ll do when determining the reachability for multimodal nodes, but I’d figure it’s the same as regular nodes. Then it might be that your transit didn’t connect well to OSM..

**Reply by kevinkreiser** [reply]
this could only happen if your transit is connected to very small osm subnetwork/island. in a well connected network lowering this number gives worse results. I guess the trick here is that reachability isn't considering transit edges. in my opinion we are better off doing a better job connecting transit to osm than having reachability consider transit edges

**Reply by moss-xyz** [reply]
Ah no, sorry @nilsnolde, I actually did mean the bicycle mode! Apologies, I've been jumping between different workstreams and I realize it is confusing.

I think I understand what you mean @kevinkreiser. Would the same apply to a bicycle network? I have no idea how the edges are connected / which edges are even considered in the network. The only confusing thing is that it works for both ped and auto, just not bike - I thought bikes would be able to use roads or sidewalks, so long as they are not explicitly marked as "not for bicycle use" in OSM?

**Comment by khamaileon** [comment]
Ah yes, thank you. I was looking in the config file. I'd forgotten about the costing options!

I hadn't looked at the time part yet. If I understand what you're saying, there's no way to specify a departure time, let alone an arrival time. But how is the graph constructed, is there a specific time used?

Regarding loading time, I'll have to try again with docker-valhalla. I have the impression that my custom build is much slower 😅

**Reply by nilsnolde** [reply]
> If I understand what you're saying, there's no way to specify a departure time, let alone an arrival time. But how is the graph constructed, is there a specific time used?

Hm, not sure we're talking about the same things:) I was referring to the web app not having anything for date/time input, simply  bcs the public FOSSGIS instance doesn't have GTFS loaded (or traffic), but it's still on the roadmap of course.

Valhalla has extensive time costing options, but indeed for multimodal we don't have arrival time, only departure time.

**Reply by khamaileon** [reply]
Thank you it's clearer to me now. Concerning the arrival time, do you know if it's something planned?

**Reply by nilsnolde** [reply]
Let's say it's smth we'd like to have as a project. As "planned" I can only speak for our company and we have no direct interest, so we won't fund this (can't speak for others, though there's very few others lately). We're always open to work on features/bugs where someone else picks up the tab though:)

**Reply by khamaileon** [reply]
Thanks for the answer. For the moment it's a side project for me, but if it works out I'll know which door to knock on 😉
